server/schedulers/utils.go

```diff
@@ -346,14 +333,29 @@ func (infl Influence) add(rhs *Influence, w float64) Influence {
 
 // TODO: merge it into OperatorInfluence.
 type pendingInfluence struct {
-	op       *operator.Operator
-	from, to uint64
-	origin   Influence
+	op           *operator.Operator
+	from, to     uint64
+	origin       Influence
+	scheduleTime uint64
+	balanceType  BalanceType
+}
+
+func (p *pendingInfluence) isDone() bool {
+	return p.op.IsEnd()
+}
+
+func (p *pendingInfluence) isTransferLeader() bool {
+	return strings.Contains(p.op.Desc(), "transfer-leader")
```

Comment:
Is there any elegant way to implement this function? Making assumptions  about operator description is not a good idea.

Reply:
I will replace it with kind